docs/configuration.md

```diff
@@ -0,0 +1,297 @@
+  steps to create genesis data, than to create a spec. Current design
+  implies that this mechanism should be used only for mainnet where we
+  want to have full understanding of which addresses have which
+  stake. The format is demonstrated below. `hash` is the hash (Blake2b
+  256 bits) of canonically encoded `mainnet-genesis.json` file. It can
+  be computed manually using `scripts/js/genesis-hash.js` script.
+
+```
+    genesis:
+      src:
+        file: mainnet-genesis.json
+        hash: 5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb
+```
+
+* Another way is to provide a specification how to generate genesis
+  data. It allows one to specify how many nodes should have stake,
+  which bootstrap stakeholders should be used, what should be total
+  balance, etc. When node is launched with spec as genesis, it
+  automatically constructs genesis data from it. This way to specify
+  genesis is supposed to be used for everything except
+  mainnet. Example:
+
+```
+    genesis:
+      spec:
+        initializer:
+          testnetInitializer:
+            testBalance:
+              poors:        12
+              richmen:      4
+              richmenShare: 0.99
+              useHDAddresses: True
+              totalBalance: 600000000000000000
+            fakeAvvmBalance:
+              count: 10
+              oneBalance: 100000
+            distribution:
+              testnetRichmenStakeDistr: []
+            seed: 0
+        blockVersionData:
+          scriptVersion:     0
+          slotDuration:      7000
+          maxBlockSize:      2000000
+          maxHeaderSize:     2000000
+          maxTxSize:         4096 # 4 Kb
+          maxProposalSize:   700 # 700 bytes
+          mpcThd:            0.01 # 1% of stake
+          heavyDelThd:       0.005 # 0.5% of stake
+          updateVoteThd:     0.001 # 0.1% of total stake
+          updateProposalThd: 0.1 # 10% of total stake
+          updateImplicit:    10 # slots
+          softforkRule:
+            initThd:        0.9 # 90% of total stake
+            minThd:         0.6 # 60% of total stake
+            thdDecrement:   0.05 # 5% of total stake
+          txFeePolicy:
+            txSizeLinear:
+              a: 155381 # absolute minimal fees per transaction
+              b: 43.946 # additional minimal fees per byte of transaction size
+          unlockStakeEpoch: 18446744073709551615 # last epoch (maxBound @Word64)
+        protocolConstants: &dev_core_genesis_spec_protocolConstants
+          k: 2
+          protocolMagic: 55550001
+          vssMinTTL: 2
+          vssMaxTTL: 6
+        ftsSeed: "c2tvdm9yb2RhIEdndXJkYSBib3JvZGEgcHJvdm9kYSA="
+        heavyDelegation: {}
+        avvmDistr: {}
+```
+
+### Initializer
+
+There are two substantially different types of genesis spec, with the
+difference in `initializer` value. Initializer can be either
+`testnetInitializer` or `mainnetInitializer`.
```

Comment:
Again, as per a previous comment -- is the key auto-genesis convenience worth the dangers of divergence between `mainnet` and `testnet`? 

Answer to this question could have non-trivial consequences on the usefulness of `testnet` for the purpose or prototype development and testing.  cc @dcoutts

Reply:
Initializer determines balances, stakes and bootstrap stakeholders. In testnet we need differences balances. For example, we need few thousands of addresses with some balance, apart from AVVM output. In mainnet we have only AVVM output (and 3 more addresses).
This difference between mainnet and testnet is intentional.